### ISSUE TITLE
MQTT Daten

### ISSUE THREAD

**Issue by damark234** [KlausLi/EspSoyosourceController#5]
Hallo,

wollte mal fragen was über MQTT alles gesendet wird. Also Info, …

Bekomme über den Info Kanal mit angeschlossenem WR ohne Batterie z.B.:
Status(23) not valid

Kann das aber nicht einordnen. 

Grüße

**Comment by KlausLi** [comment]
auf m info topic kommt derzeit nur die offline meldung von mqtt und die onlinemeldung mit uhrzeit und ip
und falls der status nicht ausgelesen werden kann not valid bzw. was falsches zurückkommt von soyo

Das ist ne Fehlermeldung daß der status nicht gelesen werden kann, eventuell wegen nicht angeschlossener Batterie oder falsch angeschlossenem Rs485 , Spannungsversorgung 3,3 volt mit bumms ?

Ich hab das ehrlich gesagt noch nie probiert ohne batterie, ,muss i mal testen :_)

**Comment by KlausLi** [comment]
schau mal ob bei dir der soyo so wire im bild eingestellt ist  und wann nicht stells so ein und berichte bitte obs n unterschied macht nach dem du den esp neugestartet hast
wann nicht hilft nur 24 volt drauf zum testen

https://ibb.co/rw5Xt3b

cp grid  = N
Autolimit = Y

**Comment by KlausLi** [comment]
ganz vergessen 
im status extrahiere ich dann die DC zeile auf der webseite vom esp , wann status not valid  kann natürlich auch nix auf der webseite in der dc zeile abgezeigt werden (BLEIBT 0) 
da wir aber die selbe fw haben gehe ich davon aus daß es am nicht angeschl. akku liegt

**Comment by xiubu-mk** [comment]
Ich bekomme über Mqtt: Info, L1L2L3, MaxPower, NotAus, WaitSekunden

Bei Info bekomme ich Updates, bei den anderen nicht. Steuern lässt sich aber darüber. Ist das so gewollt? 

Gruß xiubu-mk

**Comment by KlausLi** [comment]
HalloJa dachte das wäre klar.Sind zum steuern.Am 03.10.2021 17:20 schrieb xiubu-mk ***@***.***>:

Ich bekomme über Mqtt: Info, L1L2L3, MaxPower, NotAus, WaitSekunden

Bei Info bekomme ich Updates, bei den anderen nicht. Steuern lässt sich aber darüber. Ist das so gewollt?

Gruß xiubu-mk

**Comment by KlausLi** [comment]
freut mich daß du's dir so genau ansiehst :-)

**Comment by xiubu-mk** [comment]
Okay, ich habe es in mein Smart Home mit eingebunden. Wollte hier die NotAus und MaxPower Funktion verwenden, da man am Gerät ja nicht so detailliert die Volt einstellen kann. Funktioniert zwar so, nur wenn ich das manuell über die Webseite ändere passt es halt nicht mehr. Ansonsten funktioniert es super.

**Comment by KlausLi** [comment]
ich weiss was du meinst 
bei buttonklick notaus weiss mqtt nix davon 

da müsste ich state & command einführen ...
lassen wirs so wie s es ist

**Comment by KlausLi** [comment]
bzw. wann  ich ein json array mache {clientD:"soyo_12345",Value:true]
button click Notaus  + publish notaus
subscribe notaus on esp <- ignore own clientid

dann würde wieder ein topic genügen 
... bin kein Freund von state &cmd topics...

**Comment by xiubu-mk** [comment]
Ja, stell ich mir auch nicht so leicht vor. Muss ja auch über das Webinterface laufen. Da ist wie es jetzt ist die wohl die beste Lösung.

**Comment by KlausLi** [comment]
du bist gut :-) , das habe ich doch vorher geschrieben :-) 

aber es geht doch ohne 2ten topic , ich bau  den mqtt publish bei button notaus, notausaus und maxpower ein

**Comment by xiubu-mk** [comment]
Okay, danke :-)

**Comment by damark234** [comment]
Das ist ne Fehlermeldung daß der status nicht gelesen werden kann, eventuell wegen nicht angeschlossener Batterie oder falsch angeschlossenem Rs485 , Spannungsversorgung 3,3 volt mit bumms ?

--> Das hatte ich nicht aber hab ich nun. 

Akku simuliere ich heute mal mit nem Netzteil. Ich informiere euch wieder.
